codegenerator/cli/templates/static/codegen/src/db/DbFunctionsImplementation.js

```diff
@@ -572,6 +502,69 @@ module.exports.deleteRolledBackEntityHistory = (
     );
   `;
 
+const Utils = require("envio/src/Utils.bs.js");
+
+module.exports.pruneStaleEntityHistory = (
+  sql,
+  entityName,
+  safeChainIdAndBlockNumberArray,
+) => {
+  const tableName = makeHistoryTableName(entityName);
+  return sql`
+  WITH first_change AS (
+    SELECT
+      MIN(serial) AS first_change_serial
+    FROM
+      public.${sql(tableName)}
+    WHERE
+      ${Utils.$$Array.interleave(
+    safeChainIdAndBlockNumberArray.map(
+      ({ chainId, blockNumber }) =>
+        sql`(entity_history_chain_id = ${chainId} AND entity_history_block_number > ${blockNumber})`,
+    ),
+    sql` OR `,
+  )}
+  ),
```

Comment:
Since this doesn't relate to a given reorged chain, we need to check each chain for changes in their respective reorg thresholds and selected the earliest change out of those.

Reply:
Now `safeChainIdAndBlockNumberArray` started make sense :)

Reply:
Any suggestions for a rename @DZakh?